pkg/kv/kv_object_format.go

```diff
@@ -0,0 +1,54 @@
+package kv
+
+import (
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+var matchers = make(map[string]protoreflect.ProtoMessage)
+var defaultMsg protoreflect.ProtoMessage = nil
+
+// Register a pb message type to parse the data, according to a path regex
+// All objects which match the path regex will be parsed as that type
+// A nil type parses the value as a plain string
+func RegisterType(pathRegexp string, pb protoreflect.ProtoMessage) {
+	matchers[pathRegexp] = pb
+}
+
+// The pb message type to parse a value, in case the path does not meet
+// any regex of the above. This is done mainly to support paths which
+// have no rules and may cause a conflict with another regex
+// Note that multiple calls ti RegisterDefaultType will override each other
+func RegisterDefaultType(pb protoreflect.ProtoMessage) {
+	defaultMsg = pb
+}
+
+func resolvePath(path string) (protoreflect.ProtoMessage, error) {
+	for pathRegexp, msg := range matchers {
+		if MatchPath(path, pathRegexp) {
+			return msg, nil
+		}
+	}
+	if defaultMsg == nil {
+		return nil, ErrNotFound
+	}
+	return defaultMsg, nil
+}
+
+func ToPrettyString(path string, protoStr []byte) (string, error) {
+	msg, err := resolvePath(path)
+	if err != nil {
+		return "", err
+	}
+
+	if msg == nil {
+		return string(protoStr), nil
+	}
```

Comment:
Maybe we should have encoded version and metadata as part of the value to make this one easy. But for now I think with a little code we can just dump the field codes, types and values we find.

Reply:
Not sure I follow - here, I identify a field that is not really a pb msg, and I dump it as a string. Is that differernt from what you suggest?

Reply:
You can ignore. I thought there `msg == nil` is unknown proto we need to decode - but it means not a proto value.